Readonly wallet

pietro909 opened the pull request:
Support Wallet restore via Public Key: it gives access to the transactions history in read-only mode.

Apps and anything requiring `IWallet.sendBitcoin` or `IWallet.settle` isn't available.

Some UI changes:

| Wallet | Apps | Vtxo | Backup |
|-----------|---------|--------|-----------|
|<img width="412" height="689" alt="image" src="https://github.com/user-attachments/assets/7c497af8-3bbc-48a5-b672-823bdbd9aeae" />|<img width="412" height="689" alt="image" src="https://github.com/user-attachments/assets/3188e59d-30bf-4fee-ab52-1fd64fefa2c4" />|<img width="412" height="689" alt="image" src="https://github.com/user-attachments/assets/65fd2e07-c2b4-4ed6-b8e9-a9f3df3cc03e" /> |<img width="412" height="689" alt="image" src="https://github.com/user-attachments/assets/18803708-a3d7-4aa1-b669-e61e8adf4f37" /> | 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Read-only wallet mode: restore with public key (npub), view balance, history, receiving addresses, and dedicated read-only UI blocks across Wallet, Settings, Advanced, Backup and Apps.
  * New ReadonlyWallet component and read-only initialization/unlock flows; UI adapts to readonly state (hides private-key actions).

* **Bug Fixes**
  * Apps tab and Send actions hidden when wallet is read-only.
  * Settings options that require a private key are excluded in readonly mode.

* **Tests**
  * New and updated end-to-end tests covering readonly restore, receive flows, and UI restrictions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pietro909 committed:
Support Wallet restore from Address alone

pietro909 committed:
Allow navigation

pietro909 committed:
Support public key in backup and restore #262

pietro909 committed:
Discriminate Readonly wallet #262

pietro909 committed:
Use reader/writer wallet #262

pietro909 committed:
Introduce wallet.isReadonly #262

pietro909 committed:
Add one basic e2e test for readonly

pietro909 committed:
Remove debugging code

pietro909 committed:
Fix formatting, address Rabbit's comments

pietro909 committed:
Disable Apps in readonly mode

pietro909 committed:
Fix page refresh

pietro909 committed:
Fix e2e tests nostr

pietro909 committed:
Remove extra log

pietro909 commented on src/screens/Settings/Vtxos.tsx:
```diff
@@ -285,7 +285,7 @@ export default function Vtxos() {
                       <Text>{prettyAgo(wallet.nextRollover)}</Text>
                     </Box>
-                    {success ? <WarningBox green text='Coins renewed successfully' /> : null}
+                    <WarningBox green text='Coins renewed successfully' />
```
Conditionally showing the message https://github.com/arkade-os/wallet/pull/277/commits/dda79378884509b25361ab89a5eda4250f9c9a03

pietro909 committed:
Fix Unlock screen to cater for readonly errors

pietro909 committed:
Thank you, rabbit

pietro909 committed:
Use nip19 npub format

pietro909 committed:
Hide Apps and Send buttons in readonly mode

bordalix commented:
Should we say something on the success page when in read-only mode?

<img width="463" height="375" alt="Screenshot 2025-12-19 at 11 33 50" src="https://github.com/user-attachments/assets/078a3c9c-715f-47d9-9deb-da33f829a4d2" />

pietro909 committed:
Use one screen for restoring wallet

pietro909 committed:
Fix stale state issues, refactor single states

pietro909 committed:
Refactor wallet init to reduce state coupling and race conditions

pietro909 committed:
Show readonly banner

pietro909 committed:
Show different success message

pietro909 committed:
Remove redundant navigation to Wallet

pietro909 committed:
Update test

pietro909 committed:
Rollback wallet navigation and Apps updates

pietro909 committed:
Fix navigation bug

pietro909 committed:
Make refresh reliable, add readonly message to transaction view

pietro909 committed:
fix formatting

pietro909 committed:
Fix password, lock, vtxos, notes pages

pietro909 committed:
Add e2e tests and minor fixes

pietro909 committed:
Address rabbit's comments

pietro909 committed:
Fix connect race condition with navigation

pietro909 commented:
Thank you @bordalix 

Fixed:
- Notes, Password Change now hidden in RO mode
- VTXOs screen not showing the success message
- A bug in navigation where Boltz swaps were not being restored
- Show the RO warning in Settings and Advanced where options are now hidden
- Addressed rabbit's comments
- Added dedicated e2e tests for RO wallet

| settings | advanced |
|---------------|-------------------|
|<img width="411" height="718" alt="Screenshot from 2026-01-10 08-13-02" src="https://github.com/user-attachments/assets/7f11353c-a50f-4778-9565-8180a6bbf6c6" />|<img width="411" height="718" alt="Screenshot from 2026-01-10 08-13-06" src="https://github.com/user-attachments/assets/78e84184-55ee-43da-b776-f74c4645fbb2" />|

louisinger reviewed (approved):
tACK
